Create a new guide for installer-based provisioning (1 guide of 3) (#3472)
theforeman/foreman-documentation: guides/common/modules/proc_creating-hosts-with-pxe-provisioning.adoc

[id="creating-hosts-with-pxe-provisioning"]
= Creating hosts with PXE provisioning

Unattended provisioning is the simplest form of host provisioning.
You enter the host details on {ProjectServer} and boot your host.
{ProjectServer} automatically manages the PXE configuration, organizes networking services, and provides the operating system and configuration for the host.

This method of provisioning hosts uses minimal interaction during the process.

> To use the CLI instead of the {ProjectWebUI}, see the xref:cli-creating-hosts-with-unattended-provisioning_{context}[].

Reviewer: ```suggestion
To use the CLI instead of the {ProjectWebUI}, see the xref:cli-creating-hosts-with-unattended-provisioning[].
```

.Procedure
include::snip_steps-create-a-host-tab-host.adoc[]
include::snip_steps-create-a-host-tab-interfaces.adoc[]
. Click the *Operating System* tab, and verify that all fields contain values.
Confirm each aspect of the operating system.
. Optional: Click *Resolve* in *Provisioning template* to check the new host can identify the right provisioning templates to use.
+
For more information about associating provisioning templates, see xref:provisioning-templates[].
ifdef::katello,satellite,orcharhino[]
. Click the *Parameters* tab, and ensure that a parameter exists that provides an activation key.
If not, add an activation key.
endif::[]
. Click *Submit* to save the host details.
+

> For more information about network interfaces, see {ManagingHostsDocURL}Adding_Network_Interfaces_managing-hosts[Adding network interfaces] in _{ManagingHostsDocTitle}_.

Reviewer: ```suggestion
For more information about network interfaces, see {ManagingHostsDocURL}configuring-network-interfaces[Configuring network interfaces] in _{ManagingHostsDocTitle}_.
```

Refs

````
$ rg configuring-network-interfaces
guides/common/modules/con_configuring-network-interfaces.adoc
1:[id="configuring-network-interfaces"]
````

which was changed recently by Maxim.

This creates the host entry and the relevant provisioning settings.
This also includes creating the necessary directories and files for PXE booting the bare-metal host.
If you start the physical host and set its boot mode to PXE, the host detects the DHCP service of {ProjectServer}'s integrated {SmartProxy}, receives HTTP endpoint of the Kickstart tree and installs the operating system.

ifdef::katello,satellite,orcharhino[]
When the installation completes, the host also registers to {ProjectServer} using the activation key and installs the necessary configuration and management tools from the {project-client-name} repository.
endif::[]

> [id="cli-creating-hosts-with-unattended-provisioning_{context}"]

Reviewer: ```suggestion
[id="cli-creating-hosts-with-unattended-provisioning"]
```

.CLI procedure
. Create the host with the `hammer host create` command:
+
[options="nowrap" subs="+quotes"]
----

> # hammer host create \

Reviewer: ```suggestion
$ hammer host create \
```

--build true \
--enabled true \
--hostgroup "_My_Host_Group_" \
--location "_My_Location_" \
--mac "_My_MAC_Address_" \
--managed true \
--name "_My_Host_Name_" \
--organization "_My_Organization_"
----

> . Ensure the network interface options are set using the `hammer host interface update` command:

Reviewer: ```suggestion
. Configure the network interface:
```

To sure about this. What do you think?

+
----

> # hammer host interface update \

Reviewer: ```suggestion
$ hammer host interface update \
```
